### DISCUSSION TITLE
Discussion: SASS 2.0 & Breaking changes (removing settings)

### DISCUSSION THREAD

**Discussion by glenn2223** [glenn2223/vscode-live-sass-compiler#197] [Polls]
So SASS 2.0 will be coming (2024 @ the earliest) and, with that, there will be a lot of breaking changes.

My main concern is that the current JS API has now been marked as deprecated in favour of a newer one - so will be removed in 2.0. By upgrading to the new defacto it will remove the following settings: (which I have avoided for now)
- liveSassCompile.settings.formats[].linefeed
- liveSassCompile.settings.formats[].indentType
- liveSassCompile.settings.formats[].indentWidth

These are not in the new API and, apparently, will never be added (see sass/dart-sass#1585)

I'd like to move to the new API sooner rather than later as it has some benefits (mainly speed related). However, I don't just want to remove it without at least opening it up to the community.

Another option - according to sass/dart-sass#1585 -  is to create a postcss plugin to do this instead. I don't really have the time to work on this and can't really spend all those hours/days working on a feature that might not have a big userbase

**Comment by thecodercoder** [comment]
I obviously can't speak for everyone, but I never use those settings, as I don't do any further work in the compiled CSS file. And I think if people are using a VS Code extension to compile their Sass files, they could use Prettier to handle formatting.

**Comment by glenn2223** [comment]
Thanks for the input! It seems that the community are agreeing with you so I will mark these settings as deprecated in the next release.

This will give users plenty of time to review alternative solutions (ready for whenever SASS 2.0 is released).

I will continue to keep this open until until it's relase (probably move it over to an actual discussion though)
